satviz-consumer/src/main/java/edu/kit/satviz/consumer/gui/visualization/VisualizationController.java

```diff
@@ -71,99 +107,181 @@ private void initialize() {
   }
 
   @FXML
   private void updateHeatmapColdColor() {
-
+    mediator.updateHeatmapColdColor(coldColorColorPicker.getValue());
   }
 
   @FXML
   private void updateHeatmapHotColor() {
-
+    mediator.updateHeatmapHotColor(hotColorColorPicker.getValue());
   }
 
   @FXML
   private void highlightVariable() {
-
+    mediator.highlightVariable(highlightVariableSpinner.getValue());
   }
 
   @FXML
   private void clearHighlightVariable() {
-
+    mediator.clearHighlightVariable();
   }
 
   @FXML
   private void screenshot() {
-
+    mediator.screenshot();
   }
 
   @FXML
-  private void openScreenShotFolder() {
-
+  private void openScreenshotFolder() {
+    try {
+      Desktop.getDesktop().open(new File(ConsumerConfig.DEFAULT_SCREENSHOT_FOLDER));
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
   }
 
   @FXML
   private void startOrStopRecording() {
-
+    mediator.startOrStopRecording();
+    recording = !recording;
+    updateRecordingDisplay();
   }
 
   @FXML
   private void pauseOrContinueRecording() {
-
+    mediator.pauseOrContinueRecording();
+    recordingPaused = !recordingPaused;
+    updateRecordingPausedDisplay();
   }
 
   @FXML
   private void pauseOrContinueVisualization() {
-
+    mediator.pauseOrContinueVisualization();
+    visualizationRunning = !visualizationRunning;
+    updateVisualizationRunningDisplay();
   }
 
   @FXML
   private void relayout() {
-
+    mediator.relayout();
   }
 
   @FXML
-  private void seekToUpdate() {
+  private void updateReceivedClausesSpinner() {
+    long currentUpdate = receivedClausesSpinner.getValue();
+    receivedClausesSlider.setValue(currentUpdate);
+    mediator.seekToUpdate(currentUpdate);
+  }
 
+  @FXML
+  private void updateReceivedClausesSlider() {
+    long currentUpdate = (long) receivedClausesSlider.getValue();
+    receivedClausesSpinner.getValueFactory().setValue((int) currentUpdate);
+    mediator.seekToUpdate(currentUpdate);
   }
 
   // METHODS (OTHER)
 
+  /**
+   * A callback method to update the {@link Spinner} and {@link Slider} of the GUI
+   * for "time" controls. It is supposed to be called whenever the {@link Mediator} object
+   * of this class receives or processes new clauses.
+   *
+   * @see VisualizationController#VisualizationController(Mediator, ConsumerConfig, int)
+   * @see Mediator
+   */
   public void onClauseUpdate() {
+    long totalUpdates = mediator.numberOfUpdates();
+    long currentUpdate = mediator.currentUpdate();
+
+    // update spinner
+    receivedClausesSpinner.getEditor().textProperty().removeListener(
+        receivedClausesSpinnerIntegerValidationListener);
+    receivedClausesSpinnerIntegerValidationListener = GuiUtils.initializeIntegerSpinner(
+        receivedClausesSpinner,
+        MIN_RECEIVED_CLAUSES,
+        (int) totalUpdates,
+        (int) currentUpdate);
+
+    // update slider
+    receivedClausesSlider.setMax(totalUpdates);
+    receivedClausesSlider.setValue(currentUpdate);
```

Comment:
Dieser Teil muss in ein `Platform.runLater(() -> { ... });`, da GUI Updates auf dem JavaFX application thread ausgeführt werden müssen.